src/sql/src/plan/scope.rs

```diff
@@ -283,10 +283,21 @@ impl Scope {
             .all_items(outer_scopes)
             .filter(|(column, lat_level, item)| (matches)(*column, *lat_level, item));
         match results.next() {
-            None => Err(PlanError::UnknownColumn {
-                table: table_name.cloned(),
-                column: column_name.clone(),
-            }),
+            None => {
+                let similar = self
+                    .all_items(outer_scopes)
+                    .filter_map(|(_column, _lat_level, item)| {
+                        item.column_name
+                            .is_similar(column_name)
+                            .then(|| item.column_name.clone())
+                    })
+                    .collect();
```

Comment:
This is the part I'm more unsure about, but I think is correct

Reply:
I don't think this is correct in the general case. Not all items in the scope are visible depending on the query, which is why we need the `matches` input above. Otherwise `.filter(|(column, lat_level, item)| (matches)(*column, *lat_level, item));` could just be `.filter(|(column, lat_level, item)| item.column_name == column_name);`

So for example, `similar` may return a column name that is exactly equal to `column_name` but isn't actually visible due to some SQL rules.

Reply:
Good call, I updated the passed in closures to not check column_name, we do that separately in the function. Now we can filter to all visible objects, and check for similar column names against that